opensearch-operator/pkg/helpers/reconcile-helpers.go

```diff
@@ -103,21 +110,6 @@ func ResolveImage(cr *opsterv1.OpenSearchCluster, nodePool *opsterv1.NodePool) (
 		}
 	}
 
-	// If a custom image is specified, use it.
-	if cr.Spec.General.ImageSpec != nil {
-		if cr.Spec.General.ImageSpec.ImagePullPolicy != nil {
-			result.ImagePullPolicy = cr.Spec.General.ImageSpec.ImagePullPolicy
-		}
-		if len(cr.Spec.General.ImageSpec.ImagePullSecrets) > 0 {
-			result.ImagePullSecrets = cr.Spec.General.ImageSpec.ImagePullSecrets
-		}
-		if cr.Spec.General.ImageSpec.Image != nil {
-			// If image is set, nothing else needs to be done
-			result.Image = cr.Spec.General.ImageSpec.Image
-			return
-		}
-	}
-
```

Comment:
Since this code snippet is used 3 times in this go file, refactoring the code to call `useCustomImage` func for easy maintenance going forward.